components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/impl/OAuthJwtAuthenticatorImpl.java

```diff
@@ -145,13 +145,25 @@ private boolean handleScopeValidation(Message message, SignedJWTInfo signedJWTIn
         String scopeClaim = signedJWTInfo.getJwtClaimsSet().getStringClaim(JwtTokenConstants.SCOPE);
         if (scopeClaim != null) {
             String orgId = RestApiUtil.resolveOrganization(message);
+            if (orgId == null) {
+                log.error("Organization is not present in the request");
+                return false;
+            }
             String[] scopes = scopeClaim.split(JwtTokenConstants.SCOPE_DELIMITER);
             oauthTokenInfo.setScopes(scopes);
             String orgHandle = getOrgHandleFromJwt(signedJWTInfo);
             if (orgHandle != null) {
                 message.put(ORGANIZATION_HANDLE, orgHandle);
             }
-
+            // check whether organization claim value and orgId matches
+            JSONObject orgClaim = signedJWTInfo.getJwtClaimsSet().getJSONObjectClaim(JwtTokenConstants.ORGANIZATION);
+            if (orgClaim != null) {
+                String orgUuid = orgClaim.getAsString(JwtTokenConstants.UUID);
+                if (orgUuid != null && (!orgId.equals(orgUuid))) {
+                    log.error(String.format("Requested OrgId (%s) and the token's organization uuid (%s) mismatch!", orgId, orgUuid));
+                    return false;
+                }
+            }
```

Comment:
org id must be there in the JWT. We should not continue if its not there.
9.2.x implementation for a reference:
https://github.com/wso2/carbon-apimgt/blob/9.2.x/components/apimgt/org.wso2.carbon.apimgt.rest.api.common/src/main/java/org/wso2/carbon/apimgt/rest/api/common/utils/JWTUtil.java#L86

Reply:
Fixed via https://github.com/wso2/carbon-apimgt/pull/11670/commits/1f32adaf0ff3f026ad4d2a1312b5b1f76a7bfb93